Add fashion/clothing icons

danielbayley opened the pull request:
Related to #1520. Closes #978.

- [x] pants (`trousers`)
- [x] shorts
- [x] dress
- [x] skirt
- [x] jacket
- [x] sweater
- [x] purse (`bag-hand` (handbag) moved to #1520)
- [x] hats (also `hat-beanie` in [#1435](https://github.com/lucide-icons/lucide/pull/1435#issuecomment-1637076021))
- [x] footwear
- [x] socks

karsa-mistmere commented:
Nice icons, but some are a bit too large, especially vest and trousers, and the shirt sleeves are maybe a bit too dense.

danielbayley commented:
> Nice icons, but some are a bit too large, especially vest and trousers, and the shirt sleeves are maybe a bit too dense.

@karsa-mistmere Think I have addressed most of those points now… see what you think anyway.

ericfennis commented:
This PR is merged to the lab branch. This PR does not match our [PR template](https://github.com/lucide-icons/lucide/blob/main/.github/pull_request_template.md), which requires valid use cases. To move on we merged this PR to the [lab branch](https://github.com/lucide-icons/lucide/tree/lab), which will be moved to a separate repo in the future.
